bitcoin/examples/ecdsa-psbt-simple.rs

```diff
@@ -0,0 +1,246 @@
+use bitcoin::hashes::Hash;
+use bitcoin::locktime::absolute;
+use bitcoin::psbt::{Input, PsbtSighashType};
+use bitcoin::secp256k1::{Secp256k1, Signing};
+use bitcoin::{
+    consensus, transaction, Address, Amount, Network, OutPoint, Psbt, ScriptBuf, Sequence,
+    Transaction, TxIn, TxOut, Txid, WPubkeyHash, Witness,
+};
+
+// The master xpriv, from which we derive the keys we control.
+const XPRIV: &str = "xprv9tuogRdb5YTgcL3P8Waj7REqDuQx4sXcodQaWTtEVFEp6yRKh1CjrWfXChnhgHeLDuXxo2auDZegMiVMGGxwxcrb2PmiGyCngLxvLeGsZRq";
+
+// The derivation path for the keys we control.
+// This follows the BIP 84 derivation path for Bitcoin.
+const BIP84_DERIVATION_PATH: &str = "m/84'/0'/0'";
+
+// The master fingerprint of the master xpriv.
+const MASTER_FINGERPRINT: &str = "9680603f";
+
+// The dummy UTXO amounts we are spending.
+const DUMMY_UTXO_AMOUNT_INPUT_1: Amount = Amount::from_sat(20_000_000);
+const DUMMY_UTXO_AMOUNT_INPUT_2: Amount = Amount::from_sat(10_000_000);
+
+// The amounts we are sending to someone, and receiving back as change.
+const SPEND_AMOUNT: Amount = Amount::from_sat(25_000_000);
+const CHANGE_AMOUNT: Amount = Amount::from_sat(4_990_000); // 10_000 sat fee.
+
+// Derive the external address xpriv.
+fn get_external_address_xpriv<C: Signing>(
+    secp: &Secp256k1<C>,
+    master_xpriv: Xpriv,
+    index: u32,
+) -> Xpriv {
+    let derivation_path =
+        DerivationPath::from_str(BIP84_DERIVATION_PATH).expect("valid derivation path");
+    let child_xpriv = master_xpriv.derive_priv(secp, &derivation_path).expect("valid child xpriv");
+    let external_index = ChildNumber::from_normal_idx(0).unwrap();
+    let idx = ChildNumber::from_normal_idx(index).expect("valid index number");
+
+    child_xpriv.derive_priv(secp, &[external_index, idx]).expect("valid xpriv")
+}
+
+// Derive the internal address xpriv.
+fn get_internal_address_xpriv<C: Signing>(
+    secp: &Secp256k1<C>,
+    master_xpriv: Xpriv,
+    index: u32,
+) -> Xpriv {
+    let derivation_path =
+        DerivationPath::from_str(BIP84_DERIVATION_PATH).expect("valid derivation path");
+    let child_xpriv = master_xpriv.derive_priv(secp, &derivation_path).expect("valid child xpriv");
+    let internal_index = ChildNumber::from_normal_idx(1).unwrap();
+    let idx = ChildNumber::from_normal_idx(index).expect("valid index number");
+
+    child_xpriv.derive_priv(secp, &[internal_index, idx]).expect("valid xpriv")
+}
+
+// The address to send to.
+fn receivers_address() -> Address {
+    Address::from_str("bc1q7cyrfmck2ffu2ud3rn5l5a8yv6f0chkp0zpemf")
+        .expect("a valid address")
+        .require_network(Network::Bitcoin)
+        .expect("valid address for mainnet")
+}
+
+// The dummy unspent transaction outputs that we control.
+fn dummy_unspent_transaction_outputs() -> Vec<(OutPoint, TxOut)> {
+    let script_pubkey_1 = Address::from_str("bc1qrwuu3ydv0jfza4a0ehtfd03m9l4vw3fy0hfm50")
+        .unwrap()
+        .assume_checked()
```

Comment:
```suggestion
        .expect("a valid address")
        .require_network(Network::Bitcoin)
        .expect("valid address for mainnet")
```

I don't think we should be encouraging usage of `assume_checked` outside of test code.